presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergCommonModule.java

```diff
@@ -213,6 +215,26 @@ public StatisticsFileCache createStatisticsFileCache(IcebergConfig config, MBean
         return statisticsFileCache;
     }
 
+    @Singleton
+    @Provides
+    public ManifestFileCache createManifestFileCache(IcebergConfig config, MBeanExporter exporter)
+    {
+        CacheBuilder<ManifestFileCacheKey, ManifestFileCachedContent> delegate = CacheBuilder.newBuilder()
```

Comment:
Two Questions here -
1. Did we consider using the Caffeine cache itself for Manifest file caching? Could Caffeine Window TinyLfu eviction be effective here?
2. For the cache key, we can just use the Manifest file path string, or are there issues with that?

Reply:
I opted to create a new type for the cache key even though it's just the file in case we need to modify it more easily in the future

As for caffeine, I spoke about this with Anant, but it's more complicated to use them due to a lack of utility classes and existing methods for Jmx metric support. It's a large enough effort that moving to Caffeine should be a separate PR. I have a draft up already for this. However, it still needs a little bit of work before it's ready to review #24608